contracts/UniswapV3Staker.sol

```diff
@@ -266,6 +266,22 @@ contract UniswapV3Staker is IUniswapV3Staker, Multicall {
         emit RewardClaimed(to, reward);
     }
 
+    function claimRewardAmount(
+        IERC20Minimal rewardToken,
+        address to,
+        uint256 amountRequested
+    ) external override returns (uint256 reward) {
+        reward = (amountRequested > 0 &&
+            amountRequested <= rewards[rewardToken][msg.sender])
+            ? amountRequested
+            : rewards[rewardToken][msg.sender];
+
+        rewards[rewardToken][msg.sender] -= reward;
+        TransferHelper.safeTransfer(address(rewardToken), to, reward);
+
+        emit RewardClaimed(to, reward);
```

Comment:
Should we update this event with `amountRequested` now that they can specify the reward they can specify the reward amount?

Reply:
I think `amountRequested` should be pulled from traces